### ISSUE TITLE
multiple lists render incorrectly

### ISSUE THREAD

**Issue by konobi** [MacDownApp/macdown#640]
I have rendering issues with the following:

```markdown
# Carp 
<sub><sup>[#article](https://www.linux.com/news/carp-your-way-high-availability)</sup></sub>

* `net.inet.carp.allow`
* `net.inet.carp.arpbalance`
* `net.inet.carp.log`
* `net.inet.carp.preempt`


* `advbase`
* `advskew`
 * `rachael# ifconfig carp0 create`
 * `rachael# ifconfig carp0 vhid 1 pass tyrell 192.168.0.7`


* `/etc/hostname.carp0`
 * `up vhid 1 pass tyrell 192.168.0.7`

* `/etc/sysctl.conf`
 * `rachael# sysctl -w net.inet.carp.preempt=1`
 * `pris# ifconfig carp0 advskew 100`
```

The spacing between the list items in the lists after the first list is way too wide. Should be able to see it for yourself, if not, let me know and I'll add a screenshot. Reporting version: `Macdown: Version 0.6 (749)`

**Comment by Jmuccigr** [comment]
Here's a simpler case:

```
* one
* two
* three
* four



* new
```
Insert a comment (which isn't displayed) after the first list and the problem will disappear.

Note how github renders it, with the same problem:

* one
* two
* three
* four



* new

- - - - - 
Here's a screenshot:

![screenshot 2016-08-14 09 30 12](https://cloud.githubusercontent.com/assets/615115/17649362/bf846e20-6201-11e6-9218-b2e3802d83aa.png)

**Comment by franklinyu** [comment]
@Jmuccigr Are you trying to create a new list with three blank line? According to [CommonMark](http://spec.commonmark.org/0.26/#lists), this shouldn't interrupt the original list. This works:

```markdown
* one
* two
* three

Below is a new list.

* one again
* two again
* three again
```

**Comment by Jmuccigr** [comment]
I think the point is that the spacing is too large between three and four when four is itself followed by a blank line or two. Switch to numbered lists and you can see that the list is intact, but the spacing is odd:

```
1. one
1. two
1. three
1. four  
  
1. new
```

becomes

1. one
1. two
1. three
1. four  
  
1. new

where, to be clear, thereʻs extra space between three and four.

Looking at the [markdown syntax](http://daringfireball.net/projects/markdown/syntax#list) though, I think this is the expected behavior and itʻs from this bit:

> If list items are separated by blank lines, Markdown will wrap the items in <p> tags in the HTML output.

So four gets wrapped in <p>, even though itʻs followed and not preceded by a blank line. pandoc does this too.

**Comment by franklinyu** [comment]
The original Markdown syntax missed a lot of corner cases like this; that's why I suggested to refer to [CommonMark](http://commonmark.org/).

**Comment by Jmuccigr** [comment]
Yes, in the pandoc output from common mark, all the items get wrapped with <p> when there's a single blank line in the list, instead of just some of them. It seems that the observed behavior is considered correct for non-Commonmark though.

**Comment by franklinyu** [comment]
Well, for non-CommonMark, there are a lot of "correct" behavior. Authors of every editor/renderer just implemented what they consider the correct behavior. I am considering to open another issue to discuss whether we should move our syntax to (a super set of) CommonMark, so that we actually have something to refer to. @Jmuccigr @konobi any ideas?

**Comment by franklinyu** [comment]
@Jmuccigr Sorry, I misinterpreted what you meant until I reviewed your comments today. Now I fully agree with you: given

```markdown
1. one
1. two
1. three
1. four  

1. new
```

MacDown should have rendered

```html
<ol>
  <li>
    <p>one</p>
  </li>
  <li>
    <p>two</p>
  </li>
  <li>
    <p>three</p>
  </li>
  <li>
    <p>four</p>
  </li>
  <li>
    <p>new</p>
  </li>
</ol>
```

but now it rendered

```html
<ol>
  <li>one</li>
  <li>two</li>
  <li>three</li>
  <li>
    <p>four  </p>
  </li>
  <li>
    <p>new</p>
  </li>
</ol>
```

instead. This behavior is reproduced with Hoedown.

"Expected behavior" concluded from [CommonMark](http://spec.commonmark.org/0.26/#loose):

> A list is loose *if any of its constituent list items are separated by blank lines*, or if any of its constituent list items directly contain two block-level elements with a blank line between them.

so "being loose" is not a property of a list item, but the entire list.
